core/src/main/java/com/exadel/core/services/impl/ShopContentServiceImpl.java

```diff
@@ -82,4 +88,21 @@ public void createNewProduct(Product product) {
             log.error(String.format("ShopContentServiceImpl: %s", e.getMessage()));
         }
     }
+
+    /**
+     * Creates new tags in namespace brand
+     */
+    @Override
+    public void createNewTag(String name) {
+        try (ResourceResolver resourceResolver = ResourceResolverUtil.getResourceResolver(resolverFactory)) {
+            TagManager tagManager = resourceResolver.adaptTo(TagManager.class);
+            if (tagManager.resolve(TAG_PATH + name) == null) {
+                tagManager.createTag(TAG_PATH + name, name, "Product Brand Tag", true);
+            }
+        } catch (LoginException e) {
+            log.error(String.format("createNewTag: %s", e.getMessage()));
```

Comment:
Please use placeholders instead of string formatting and always log exception objectlog.error("createNewTag: {}", name, e);
Print of 'e' will contain e.getMessage()